backend/app/services/knowledge/permission_request_service.py

```
@@ -0,0 +1,630 @@
+                Permission.user_id == user_id,
+            )
+            .first()
+        )
+
+        if (
+            existing_permission
+            and existing_permission.status == PermissionStatus.PENDING
+        ):
+            raise ValueError(
+                "You already have a pending request for this knowledge base"
+            )
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)